code/MANAGER/MANAGER.py

```diff
@@ -1,38 +1,140 @@
 
 
-# #######################              ENUMS               #######################
 
-# #######################              CLASSES             #######################
+#######################              CLASSES             #######################
 
 class MANAGER_c():
+    '''Class to manage the state machine'''
     def __init__(self) -> None:
-        self._epc_config: STM_FLASH_Epc_Conf_c = None
-        self._status: TERM_Option_e = None
+        self.__epc_config: STM_FLASH_Epc_Conf_c = None
+        self.__status: TERM_Option_e = TERM_Option_e.CONF_DEV
+        self.__power = PWR_c(config_path_file = f"") #TODO: check path
+    
+    
+    def _calibStatus(self, mode: PWR_Mode_e) -> None:
+        '''Calibrate the EPC in the specified mode
+
+        Args:
+            - mode (PWR_Mode_e): Mode to calibrate
+        Returns:    
+            - None
+        Raises:
+            - None
+        '''
+        if mode is PWR_Mode_e.VOLT_HS:
+            log.info(f"Calibrating voltage high side")
+            self.__power.calibVoltHS
+        elif mode is PWR_Mode_e.VOLT_LS:
+            log.info(f"Calibrating voltage low side")
+            self.__power.calibVoltLS
+        elif mode is PWR_Mode_e.CURR:
+            log.info(f"Calibrating current")
+            self.__power.calibCurr
+        elif mode is PWR_Mode_e.TEMP_ANOD:
+            log.info(f"Calibrating temperature anode")
+            pass
+        elif mode is PWR_Mode_e.TEMP_AMB:
+            log.info(f"Calibrating temperature ambient")
+            pass
+        elif mode is PWR_Mode_e.TEMP_BODY:
+            log.info(f"Calibrating temperature body")
+            pass
+
+
+    def executeMachineStatus(self) -> None:
+        '''Execute the state machine
+        Args:
+            - None
+        Returns:
+            - None
+        Raises:
+            - None
+        '''
+        check_calib = {PWR_Mode_e.VOLT_HS: False, PWR_Mode_e.VOLT_LS: False, PWR_Mode_e.CURR: False,
+                       PWR_Mode_e.TEMP_ANOD: False, PWR_Mode_e.TEMP_AMB: False, PWR_Mode_e.TEMP_BODY: False}
+        guided = False
+        term = TERM_c()
+
+        while self.__status is not TERM_Option_e.EXIT:
+            #Guided mode option
+            if guided is True:
+                option_guided = [None, TERM_Option_e.FLASH_ORIG, TERM_Option_e.CONF_DEV, TERM_Option_e.CALIB, TERM_Option_e.FLASH_CALIB, TERM_Option_e.EXIT]
+                self.__status = option_guided[option_guided.index(self.__status) + 1]
+            else:
+                self.__status = term.queryOptions()
+
+            #Flash original program
+            if self.__status is TERM_Option_e.FLASH_ORIG:
+                log.info(f"Flashing original program")
+                pass
 
-    def _calibStatus(self):
-        pass
+            #Configure device
+            elif self.__status is TERM_Option_e.CONF_DEV:
+                log.info(f"Configuring device")
+                self.__epc_config = term.queryEPCConf()

```

Comment:
Add `ConfigureDevice` from power

Reply:
Use CONFIG_WS_C.UpdatePath when device info is changed

Reply:
Done.